File: patienthistory/src/main/java/com/abernathyclinic/patienthistory/service/NoteService.java
package com.abernathyclinic.patienthistory.service;

import java.util.List;

import com.abernathyclinic.patienthistory.dto.NoteDTO;
import com.abernathyclinic.patienthistory.exception.NoteNotFoundException;

public interface NoteService {

	List<NoteDTO> getAllNotes();

	NoteDTO getNoteById(String id) throws NoteNotFoundException;

	List<NoteDTO> getNoteByPatientId(long patientId) throws NoteNotFoundException;

	NoteDTO saveNote(NoteDTO newNote);

	NoteDTO updateNote(String id, NoteDTO modNote) throws NoteNotFoundException;

	void deleteNote(String id) throws NoteNotFoundException;

}

File: abernathysui/src/app/services/note.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Note } from '../models/note.model';
import { map, Observable, switchMap } from 'rxjs';
import { environment } from 'src/environments/environment';

@Injectable({
    providedIn: 'root'
  })
export class NoteService {

    constructor(private http: HttpClient){}

      getAllNotes(): Observable<Note[]>{
        return this.http.get<Note[]>(environment.noteHost+'/notes');
      }

      getNoteById(noteId: string): Observable<Note> {
        return this.http.get<Note>(environment.noteHost+`/note/${noteId}`);
      }

      updateNote(noteId: string, formValue: {createdAt: Date, patientId: number, patientName: string, recommendation: string}): Observable<Note> {
        let modNote: Note = {...formValue, id: noteId, updatedAt: null};
        return this.http.put<Note>(environment.noteHost+`/note/update/${noteId}`, modNote);
      }

      addNote(formValue: {patientId: number, patientName: string, recommendation: string}): Observable<Note> {
        let newNote: Note = {...formValue, id: null, createdAt: null, updatedAt: null};  
        return this.http.post<Note>(environment.noteHost+`/note/add`, newNote);
        }

      getNotesByPatientId(patientId: number): Observable<Note[]> {
        return this.http.get<Note[]>(environment.noteHost+`/notes/${patientId}`)
      }

      deleteNote(noteId: string | null){
        return this.http.delete(environment.noteHost+`/note/delete/${noteId}`);
      }

}